data_safe_haven/commands/deploy.py

```diff
@@ -77,7 +74,9 @@ def shm(
     except DataSafeHavenError as exc:
         msg = f"Could not deploy Data Safe Haven Management environment.\n{exc}"
         raise DataSafeHavenError(msg) from exc
-
+    finally:
+        # Upload Pulumi config to blob storage
+        pulumi_config.upload(context)
```

Comment:
I can't remember if there are tests for the deploy command but we should add one to check that upload is always called whether or not an exception is raised.

Reply:
Can you add this or open an issue?

Reply:
Done in [db1fc3d](https://github.com/alan-turing-institute/data-safe-haven/pull/1839/commits/db1fc3d40bea75dfd850c4e7dd0cf1ed4a4277ae)